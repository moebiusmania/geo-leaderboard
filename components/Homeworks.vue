<script setup lang="ts">
const { data: homeworks } = defineProps<{
  data: Array<string>;
}>();
</script>

<template>
  <section class="paper">
    <div class="content">
      <h3>ðŸ“š Compiti per casa</h3>
      <p>
        Ecco gli esercizi da completare per questa stagione. Completa tutti gli
        esercizi per ottenere il massimo dei punti!
      </p>
      <ul>
        <li v-for="(homework, index) in homeworks" :key="index">
          <a :href="homework" target="_blank" rel="noopener">
            Esercizio {{ index + 1 }}
          </a>
        </li>
      </ul>
    </div>
  </section>
</template>

<style scoped>
.paper {
  background: #fff;
  padding: 2rem;
  border-radius: 2px;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 10px 0 -5px #eee,
    0 10px 1px -4px rgba(0, 0, 0, 0.15), 0 20px 0 -10px #eee,
    0 20px 1px -9px rgba(0, 0, 0, 0.15);
  max-width: 600px;
  margin: 5rem auto 0;
}

.content {
  & h3 {
    font-family: var(--title-font);
    color: var(--primary);
    margin: 0 0 1rem;
    font-size: 1.8rem;
  }

  & p {
    margin: 0 0 1.5rem;
    line-height: 1.5;
  }

  & ul {
    margin: 0;
    padding: 0;
    list-style: none;

    & li {
      margin: 0.5rem 0;
      padding-left: 1rem;
      border-left: 3px solid var(--primary);

      & a {
        color: var(--text);
        text-decoration: none;
        font-weight: 600;
        transition: color 0.2s ease;

        &:hover {
          color: var(--primary);
        }
      }
    }
  }
}
</style>
