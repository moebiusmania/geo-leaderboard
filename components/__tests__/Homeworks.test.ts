import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Homeworks from "../Homeworks.vue";

describe("Homeworks", () => {
  const mockHomeworks = [
    {
      label: "Homework 1",
      url: "https://example.com/homework1",
    },
    {
      label: "Homework 2",
      url: "https://example.com/homework2",
    },
    {
      label: "Homework 3",
      url: "https://example.com/homework3",
    },
  ];

  it("renders properly with homework list", () => {
    const wrapper = mount(Homeworks, {
      props: {
        data: mockHomeworks,
      },
    });

    // Create snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});
