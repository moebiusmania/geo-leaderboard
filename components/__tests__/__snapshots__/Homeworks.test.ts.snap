// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homeworks > renders properly with homework list 1`] = `
"<section data-v-29ac6caa="" class="paper">
  <div data-v-29ac6caa="" class="content">
    <h3 data-v-29ac6caa="">ðŸ“š Compiti per casa</h3>
    <p data-v-29ac6caa=""> Ecco gli esercizi da completare per questa stagione. Completa tutti gli esercizi per ottenere il massimo dei punti! </p>
    <ul data-v-29ac6caa="">
      <li data-v-29ac6caa=""><a data-v-29ac6caa="" href="https://example.com/homework1" target="_blank" rel="noopener">Homework 1</a></li>
      <li data-v-29ac6caa=""><a data-v-29ac6caa="" href="https://example.com/homework2" target="_blank" rel="noopener">Homework 2</a></li>
      <li data-v-29ac6caa=""><a data-v-29ac6caa="" href="https://example.com/homework3" target="_blank" rel="noopener">Homework 3</a></li>
    </ul>
  </div>
</section>"
`;
